### ISSUE TITLE
Feature request: trackball mode

### ISSUE THREAD

**Issue by ashelleyPurdue** [Ryochan7/DS4Windows#85]
The Steam Controller has a feature called "trackball mode", which gives the mouse cursor some momentum when the user flicks the trackpad.  This feature would be particularly useful with the DS4, where your right doesn't have as much room to move as it does on the steam controller.

If nobody is feeling up to it, I'm willing to implement it, since it's a simple addition.  I already have a prototype working on my other computer.

EDIT: I've forked this project and made a working prototype:
 https://github.com/ashelleyPurdue/DS4Windows/tree/touchpad_trackball

I'd submit a pull request, but it's not ready to be merged into master yet(it can't be toggled in the UI yet).

**Comment by Ryochan7** [comment]
That might be an interesting addition but I am not sure how useful it would be for the DS4. The touchpad is not really in a decent position on the controller to be able to use it as a primary control mechanism. Having trackball mode for the DS4 touchpad doesn't seem like it would offer that much more utility. I will take a look at your fork and test it out.

**Comment by Ryochan7** [comment]
Is trackball mode supposed to work like the Steam Controller? I was expecting for the cursor to keep moving after swiping and untouching the touchpad. As it is, it just seems like there is a large amount of smoothing enabled and the cursor stops once the touchpad is no longer touched. I would not expect trackball mode to be active while the touchpad is still being touched.

**Comment by Ryochan7** [comment]
Made sure to build from the proper branch this time. Trackball mode is working mostly as expected but the cursor keeps going to the upper left corner no matter which direction that the cursor was initially traveling.

**Comment by ashelleyPurdue** [comment]
How can I make a proper branch on this project?

**Comment by Ryochan7** [comment]
Making a separate branch is the better way to go. I didn't look thoroughly to see that the changes were in a new branch. When I performed a checkout of the repository, I only pulled from the jay branch rather than the touchpad_trackball branch. This won't be a problem when a pull request is made.

**Comment by Ryochan7** [comment]
Some testing has gone into implementing a trackball mode. I like the trackball emulation used in SC-Controller so I started looking into how it handles trackball emulation. It took some time to understand the rotation routines that were used when creating the various variables. The prototype design and code deviates from SC-Controller and it performs a bit differently. With default friction enabled, trackball mode is a little looser than using Medium trackball friction on a Steam Controller when using the official Steam controller mapper. More work has to be done before there is any chance of trackball emulation being included in DS4Windows.

https://gist.github.com/Ryochan7/ab233ea1ae37fd5dba287d97b668e945

**Comment by Ryochan7** [comment]
Although the settings might be tweaked later, the current implementation works well enough. Hopefully these changes improve the experience of using the touchpad for desktop and menu usage.
